modules/highgui/src/window_cocoa.mm

```diff
@@ -944,8 +944,8 @@ - (void)createSliderWithName:(const char *)name maxValue:(int)max value:(int *)v
     [[slider name] setStringValue:cvname];
     [[slider slider] setMaxValue:max];
     [[slider slider] setMinValue:0];
-    [[slider slider] setNumberOfTickMarks:(max+1)];
-    [[slider slider] setAllowsTickMarkValuesOnly:YES];
+    [[slider slider] setNumberOfTickMarks:(max > 100 ? 101 : max+1)];
+    [[slider slider] setAllowsTickMarkValuesOnly:NO];
```

Comment:
Allow selection of values without TickMarks. This is important for situations where there is a large range of values.